src/basic/path-util.c

```diff
@@ -1441,3 +1441,36 @@ int path_glob_can_match(const char *pattern, const char *prefix, char **ret) {
                 *ret = NULL;
         return false;
 }
+
+const char* default_path(void) {
+#if HAVE_SPLIT_BIN
+        static int split = -1;
+        int r;
+
+        /* Check whether /usr/sbin is separate and return the appropriate $PATH.
+         * On error assume we need split bin as configured… */
+
+        if (split < 0)
+                FOREACH_STRING(path, "/usr/sbin", "/usr/local/sbin") {
+                        _cleanup_free_ char *target = NULL;
+
+                        r = chase(path, /* root= */ NULL, /* flags= */ 0, &target, NULL);
```

Comment:
isn't chase() a bit overkill for this?

Given this is ultimately just an optimization, why not just do readlink_malloc() and compare if the result is "bin"?

Reply:
Ugh, this was just discussed and changed in https://github.com/systemd/systemd/pull/32337#discussion_r1570893430

Reply:
You're late to the party ;) That's exactly what I was arguing, but @yuwata and @YHNdnzj strongly requested `chase()`. I don't particuarly care either way, but `chase()` has the advantage that the code is a bit simpler because we don't have to worry about different ways that people could figure out to spell the path (`bin|./bin|../usr/bin|/usr/bin` to maybe even something else).

Reply:
> because we don't have to worry about different ways that people could figure out to spell the path (bin|./bin|../usr/bin|/usr/bin to maybe even something else).

Using `/proc/self/root/usr/bin` is the only correct way of course.

(Ignore the edit of Zbigniew comment, I'm too tired and accidentally hit "edit" instead of quote to post this witty comment.)

Reply:
I have do admit I didn't think about `/proc/self/root/usr/bin`… That's a nice one.